src/DataStructures/Variables.hpp

```diff
@@ -268,6 +270,13 @@ class Variables<tmpl::list<Tags...>> {
   template <typename VT, bool VF>
   Variables& operator=(const blaze::DenseVector<VT, VF>& expression);
 
+  // Prevent the previous two declarations from implicitly converting
```

Comment:
would a static_assert using is_convertible have caught this in a test?  I.e. would `static_assert(std::is_convertible<DataVector, Variables<tmpl::list<something>>>)` succeed before and fail now?

Also, how about adding an ASSERT to the converting constructors that test that `(*expression).size() % number_of_independent_components == 0)`

Reply:
The static assertion would have caught it.  Added that and a new commit for the size check.